sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -117,6 +117,21 @@ case class HashAggregateExec(
         // so return an empty iterator.
         Iterator.empty
       } else {
+        val filterPredicates = new mutable.HashMap[Int, GenPredicate]
+        aggregateExpressions.zipWithIndex.foreach{
+          case (ae: AggregateExpression, i) =>
+            ae.mode match {
+              case Partial | Complete =>
+                ae.filter.foreach { filterExpr =>
+                  val filterAttrs = filterExpr.references.toSeq
+                  val predicate = newPredicate(filterExpr, child.output ++ filterAttrs)
+                  predicate.initialize(partIndex)
+                  filterPredicates(i) = predicate
+                }
```

Comment:
Can we move this logic into `AggregationIterator`?

Reply:
Since newPredicate is a protection method for SparkPlan.

Reply:
Can we send a PR to remove `newPredicate` from `SparkPlan`? They are not related and `newPredicate` doesn't use any states of `SparkPlan`. Doing this we can simply the PR a lot.

Reply:
Ah, that's right. I'll make a pr for that.

Reply:
Fixed in https://github.com/apache/spark/pull/26604. Can you brush up this pr based on the commit?

Reply:
I just change this PR based on #26604.